http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -855,18 +855,11 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       .transaction
     _ <- wsmResources.traverse { resource =>
       for {
-        wsmState <- resourceType match {
-          case WsmResourceType.AzureDatabase =>
-            wsmClientProvider.getDatabaseState(userInfo.accessToken.token, workspaceId, resource.resourceId)
-          case WsmResourceType.AzureKubernetesNamespace =>
-            wsmClientProvider.getNamespaceState(userInfo.accessToken.token, workspaceId, resource.resourceId)
-          case WsmResourceType.AzureManagedIdentity =>
-            wsmClientProvider.getIdentityState(userInfo.accessToken.token, workspaceId, resource.resourceId)
-          case WsmResourceType.AzureDisk =>
-            wsmClientProvider.getDiskState(userInfo.accessToken.token, workspaceId, resource.resourceId)
-          case WsmResourceType.AzureStorageContainer =>
-            F.pure(WsmState(None)) // no get endpoint for a storage container in WSM yet
-        }
+        wsmState <- wsmClientProvider.getWsmState(userInfo.accessToken.token,
```

Comment:
I'm really confused how this solves the issue as the ticket is written. First of all, the ticket is talking about createApp not delete? I bet the issue exists for both though.

Secondly, there shouldn't the main logic needed be backleo logic?

As I understand it, the flow is as follows:
1. Leo recieves API request to create/delete app
2. Leo queues message to backleo
3. backleo makes various API calls to WSM to create/delete resources

As I see it, the issue is in #3. Rob's comment on the ticket supports this assessment. Atomically before we submit a request to WSM, we need to check if the state of the object permits that request (or just not fail on 409/404 from the request itself instead of making an additional WSM call which seems better). 

The main issue being that pubsub guarantees re-delivery of any message if we do not fully process it, but our backend processing can be halfway done and then a resubmission causes a second WSM call request to do the same thing which errors out. 

Am I missing something?

Reply:
this change is part of the refactor I did to how the wsmApiClientProvider structures the getWsmState calls, so not necessary for this PR (just a cleanup I initially thought was necessary for this PR, then realized it wasn't), I can split that out separately if that's helpful

Reply:
the changes that address the true problem are in the AKSinterpreter, mainly adding the retrieveWsmNamespace function and changing how similar retrieveWsm... functions are used in the createApp function

Reply:
Ohhh ok I totally missed that file, I must have collapsed it or something 😅

Reply:
no worries! sorry the changeset got a bit bloated by the other fix I added in...